pkg/storage/mvcc.go

```diff
@@ -5007,13 +5001,9 @@ func MVCCGarbageCollect(
 					// For non deletions, we need to find if we had a range tombstone
 					// between this and next value (prevNanos) to use its timestamp for
 					// computing GCBytesAge.
-					for ; rangeIdx < len(rangeTombstoneTss); rangeIdx++ {
-						rangeTS := rangeTombstoneTss[rangeIdx]
-						if rangeTombstoneTss[rangeIdx].Less(unsafeIterKey.Timestamp) {
-							break
-						}
-						if rangeTS.WallTime < fromNS {
-							fromNS = rangeTS.WallTime
+					if kv, ok := rangeTombstones.FirstAbove(unsafeIterKey.Timestamp); ok {
```

Comment:
Wrap this in a `if !rangeTombstones.IsEmpty()`, since the call has been seen to be quite expensive otherwise (even though I added a check inside `FirstAbove()` itself too).